### PR TITLE
BatchNorm2d Implementation

### PR DESCRIPTION
**Pull request by vb000**

### CODE REVIEW

File: torch/nn/modules/module.py [hunk 1]
```diff
@@ -297,7 +297,16 @@ def hammerblade(self):
         Returns:
             Module: self
         """
-        return self._apply(lambda t: t.hammerblade())
+        def apply_hb(t):
+            if isinstance(self, torch.nn.BatchNorm2d) and \
+                    (t.dtype == torch.int64):
+                # BatchNorm2d's batch counter is a long tensor. Since it is metadata
```
Comment (yodada) [inline torch/nn/modules/module.py:303]: is this the 64bit integer add we saw?

File: setup_cosim_build_env.sh [hunk 2]
```diff
@@ -44,8 +44,9 @@ else
 fi
 
 # Build COSIM runtime library and simulation executable
+export BSG_MACHINE=4x4_fast_n_fake
```
Comment (yodada) [inline setup_cosim_build_env.sh:47]: this will override the settings in `bsg_replicant/machine.mk` right?

**Review by yodada — APPROVED** [review]
Awesome! Thanks a lot for implementing this

File: torch/nn/modules/module.py (hunk 1 above)
Comment (vb000) [reply torch/nn/modules/module.py:303]: Yes, later found out this not the core tensor in the kernel computation. This is just the metadata of `torch.nn.BatchNorm2d` layer. Meaning `torch.nn.functional.batch_norm` doesn't need this.

File: setup_cosim_build_env.sh (hunk 2 above)
Comment (vb000) [reply setup_cosim_build_env.sh:47]: Yes, just wanted to be able to set the machine in pytorch, independent of what's set in bsg_replicant.
